why-conventional-commits.qmd

```diff
@@ -0,0 +1,249 @@
+rather by the team working on a given project, the question becomes:
+
+> Which commit message convention should we follow when we write commit
+> messages in Seedcase projects?
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+In the Seedcase projects, we emphasise the open-science philosophies of
+transparency and collaboration. Therefore, it is essential to have a
+clear and consistent commit message convention to maintain these
+principles, both within the team and across external contributions.
+
+Since we follow the [GitHub flow](why-github-flow.qmd), changes are made
+on topic branches and reviewed before merging into the main branch.
+Having a clear commit message convention can ease the review process, as
+the context and purpose of changes will be clearer for reviewers.
+
+In addition, some commit message conventions can automate changelogs and
+release processes, which is beneficial for version management and for
+communicating changes to users and contributors.
+
+::: callout-caution
+Note that most commit conventions and best practices follow the same
+general principles, with some differences in syntax and structure.
+
+In general, a commit message should be **clear and concise** (brief, but
+specific descriptions, avoiding vagueness and ambiguity), **relevant**
+(focus the *why*, not the *how* of the implementation), and
+**consistent** (follow a standard format).
+
+A commit message should:
+
+-   Include a short descriptive title, potentially followed by a body
+    with more details.
+-   Use the imperative, present tense in the title and the body (e.g.,
+    "Add feature" instead of "Added feature")
+-   Use the title to summarise the changes made in the commit.
+-   Use a potential body to provide context and justifications for the
+    change (such as elaboration on the changes, breaking changes,
+    reference of issues, and tagging relevant people).
+:::
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+There seem to be many general recommendations for best practices and
+fewer established conventions for how commit messages should be written.
+We will consider the conventions Conventional Commits and Gitmoji in the
+following sections.
+
+### Conventional commits
+
+The [Conventional Commits](https://www.conventionalcommits.org/)
+convention builds on the [Angular
+convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines),
+but was generalised to go beyond Typescript. It's widely adopted due to
+its compatibility with semantic versioning and automated release
+processes. Moreover, it's commonly used in various software development
+communities, including the open science communities. Its structured
```

Comment:
```suggestion
communities, including the open source communities. Its structured
```

Open source, not science right? Commit message conventions are not well used within open science unfortunately... Version control isn't as common either!

Reply:
Ah, yes! Thanks for the catch!